contracts/xGNANABribes.sol

```diff
@@ -0,0 +1,155 @@
+// SPDX-License-Identifier: GPL-3.0-only
+pragma solidity 0.8.17;
+
+/*
+  ______                     ______                                 
+ /      \                   /      \                                
+|  ▓▓▓▓▓▓\ ______   ______ |  ▓▓▓▓▓▓\__   __   __  ______   ______  
+| ▓▓__| ▓▓/      \ /      \| ▓▓___\▓▓  \ |  \ |  \|      \ /      \ 
+| ▓▓    ▓▓  ▓▓▓▓▓▓\  ▓▓▓▓▓▓\\▓▓    \| ▓▓ | ▓▓ | ▓▓ \▓▓▓▓▓▓\  ▓▓▓▓▓▓\
+| ▓▓▓▓▓▓▓▓ ▓▓  | ▓▓ ▓▓    ▓▓_\▓▓▓▓▓▓\ ▓▓ | ▓▓ | ▓▓/      ▓▓ ▓▓  | ▓▓
+| ▓▓  | ▓▓ ▓▓__/ ▓▓ ▓▓▓▓▓▓▓▓  \__| ▓▓ ▓▓_/ ▓▓_/ ▓▓  ▓▓▓▓▓▓▓ ▓▓__/ ▓▓
+| ▓▓  | ▓▓ ▓▓    ▓▓\▓▓     \\▓▓    ▓▓\▓▓   ▓▓   ▓▓\▓▓    ▓▓ ▓▓    ▓▓
+ \▓▓   \▓▓ ▓▓▓▓▓▓▓  \▓▓▓▓▓▓▓ \▓▓▓▓▓▓  \▓▓▓▓▓\▓▓▓▓  \▓▓▓▓▓▓▓ ▓▓▓▓▓▓▓ 
+         | ▓▓                                             | ▓▓      
+         | ▓▓                                             | ▓▓      
+          \▓▓                                              \▓▓         
+ * App:             https://ApeSwap.finance
+ * Medium:          https://ape-swap.medium.com
+ * Twitter:         https://twitter.com/ape_swap
+ * Telegram:        https://t.me/ape_swap
+ * Announcements:   https://t.me/ape_swap_news
+ * Reddit:          https://reddit.com/r/ApeSwap
+ * Instagram:       https://instagram.com/ApeSwap.finance
+ * GitHub:          https://github.com/ApeSwapFinance
+ */
+
+import "./xGNANA.sol";
+import "@openzeppelin/contracts/utils/structs/EnumerableSet.sol";
+import "@openzeppelin/contracts/access/AccessControlEnumerable.sol";
+
+/**
+    // TODO: Possible features
+    - [x] AccessControl to be able to whitelist IERC20VotesDelegatee from factory contracts
+    - [ ] Factory contract which can deploy bribe contracts with the callback functions below 
+    - [ ] Relevant events
+    - [ ] NatSpec comments
+    - [ ] Some ability to accumulate non GNANA revenue
+          - I would say that my argument still applies that we should perform buy backs at strategic points when there is high value
+             Is there a way we could mix in like stable revenue and with GNANA for xGNANA holders? 
+
+    - [x] Provide an admin role to be able to add new white listers
+    - [x] Be able to remove delegatee contracts
+ */
+
+import "@openzeppelin/contracts/utils/introspection/IERC165.sol";
+interface IERC20VotesDelegatee is IERC165 {
+    function syncDelegatePower(address delegator) external;
+}
+
+/**
+ * The intent here is that a factory contract could be used to deploy bridge contracts
+ * where a protocol can fill up with tokens and distribute tokens to xGNANA holders who 
+ * delegate their voting power.
+ */
+interface IxTokenWithBribes {
```

Comment:
Maybe `IxTokenDelegatorCallback` is more generic and "bribes" would still fall under this, but of course the idea is to be extensible and allow many other options.